test_cases/deploy_time_variables/contract.py

```diff
@@ -0,0 +1,11 @@
+from puyapy import BytesDeployVar, UInt64, UInt64DeployVar, arc4, Bytes
+
+
+class DeployTimeVariablesContract(arc4.ARC4Contract):
```

Comment:
Once we're happy with the awst and below implementation,

I'm envisioning class options to specify updatable/deletable with a value of `true | false | 'from_deploy_var' | None` and we will auto generate the code. `None` would mean no auto generated code and the user could write an explicit implementation. Might be worth exporting the `UPDATABLE` and `DELETABLE` variable names as constants in case someone wants an explicit implementation still governed by the variables the app clients provide

Reply:
I think this can just be a mixin contract, rather than couple it tightly into the compiler as a class option?

Reply:
Sounds good - where would we put this contract? I assume it wouldn't go in the stubs (since it would have an actual implementation? I've added constants for `TMPL_UPDATABLE` and `TMPL_DELETABLE` but these also required some compiler gymnastics to resolve in the compiler because they are only stubs.

Reply:
I'm thinking it would be in another repo / library entirely perhaps? `fpuyapy-utils` :P
We could shove `itoa` in there too haha